CMakeLists.txt

```diff
@@ -1790,11 +1799,16 @@ add_executable(${LOVE_EXE_NAME} WIN32 src/love.cpp ${LOVE_RC})
 target_link_libraries(${LOVE_EXE_NAME} ${LOVE_LIB_NAME})
 set_target_properties(${LOVE_EXE_NAME} PROPERTIES C_VISIBILITY_PRESET hidden CXX_VISIBILITY_PRESET hidden VISIBILITY_INLINES_HIDDEN ON)
 
-if(MSVC)
+if(MSVC OR MINGW)
 	add_executable(${LOVE_CONSOLE_EXE_NAME} src/love.cpp ${LOVE_RC})
 	target_link_libraries(${LOVE_CONSOLE_EXE_NAME} ${LOVE_LIB_NAME})
 endif()
 
+if(MINGW)
+	target_compile_definitions(${LOVE_EXE_NAME} PRIVATE SDL_MAIN_HANDLED)

```

Comment:
Is this actually needed? I thought it was only for situations where the app definitely doesn't want to use SDL's main entry point redirection (whereas love should use it). I think it also needs an extra function call in the main entry point if it's defined, too.

Reply:
You are right, I misunderstood the definition. I thought it would enable the SDL2 main, when in fact, it did quite the opposite. The reason I put it there was because of an undefined reference to WinMain. After [some digging](https://wiki.libsdl.org/SDL2/FAQWindows#i_get_undefined_reference_to_winmain16), I found out, that I need to link against `mingw32` and `SDL2main` before linking against `SDL2`, as you have mentioned earlier. It's now fixed in [b08aa24](
https://github.com/love2d/love/commit/b08aa244c5d5c03c3a468e77dcd58d20fcb77475#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR225-R231).